backend/app/store/image/fs_store.go

```diff
@@ -36,35 +36,39 @@ type FileSystem struct {
 	}
 }
 
-// Save data from reader for given file name to local FS, staging directory. Returns id as user/uuid.ext
-// Files partitioned across multiple subdirectories and the final path includes part, i.e. /location/user1/03/123-4567.png
-func (f *FileSystem) Save(fileName string, userID string, r io.Reader) (id string, err error) {
+// SaveWithID saves data from reader with given id
+func (f *FileSystem) SaveWithID(id string, r io.Reader) (string, error) {
 	data, err := readAndValidateImage(r, f.MaxSize)
 	if err != nil {
-		return "", errors.Wrapf(err, "can't load image %s", fileName)
+		return "", errors.Wrapf(err, "can't load image with ID %s", id)
 	}
 
-	data, resized := resize(data, f.MaxWidth, f.MaxHeight)
-
-	id = path.Join(userID, guid()) + filepath.Ext(fileName) // make id as user/uuid.ext
+	data = resize(data, f.MaxWidth, f.MaxHeight)
 	dst := f.location(f.Staging, id)
-	if resized { // resized also converted to png
-		id = strings.TrimSuffix(id, filepath.Ext(id)) + ".png"
-		dst = f.location(f.Staging, id)
-	}
 
 	if err = os.MkdirAll(path.Dir(dst), 0700); err != nil {
 		return "", errors.Wrap(err, "can't make image directory")
 	}
 
 	if err = ioutil.WriteFile(dst, data, 0600); err != nil {
-		return "", errors.Wrapf(err, "can't write image file %s", dst)
+		return "", errors.Wrapf(err, "can't write file")
 	}
 
-	log.Printf("[DEBUG] file %s saved for image %s, size=%d", dst, fileName, len(data))
+	log.Printf("[DEBUG] file %s saved for image %s, size=%d", dst, id, len(data))
 	return id, nil
 }
 
+// Save data from reader for given file name to local FS, staging directory. Returns id as user/uuid
+// Files partitioned across multiple subdirectories and the final path includes part, i.e. /location/user1/03/123-4567
+func (f *FileSystem) Save(fileName string, userID string, r io.Reader) (id string, err error) {
```

Comment:
Named return parameters `(id string, err error)` increase the clarity of signature but not used properly inside the code, I suggest swapping `id = path.Join(userID, guid())` with something like `tempID`, and renaming `finalID` to `id`.

Reply:
Fixed in #581